pkg/kubernetes/nulecule.js

```diff
@@ -0,0 +1,182 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2014 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "kubernetes/config"
+], function($, cockpit, config) {
+    "use strict";
+
+    var nulecule = { };
+    var _ = cockpit.gettext;
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "nulecule")
+            console.debug.apply(console, arguments);
+    }
+
+    function failure(ex) {
+        console.warn(ex);
+    }
+
+    /**
+     * NuleculeClient
+     *
+     */
+    function NuleculeClient() {
+        var self = this;
+        var http = cockpit.http(5000);
+
+        self.get_status = function get_status() {
+            var req = http.get("/atomicapp-run/api/v1.0/status")
+                        .done(function(data) {
+                            req = null;
+
+                            var response;
+                            try {
+                                response = JSON.parse(data);
+                                status = response.items[response.items.length-1];
+                                console.log("status = " + JSON.stringify(status));
+                                return JSON.stringify(status);
+                            } catch(ex) {
+                                debug("not an api endpoint without JSON data on:");
+                                return "ERROR: not an api endpoint without JSON data on";
+                            }
+
+                        })
+                        .fail(function(ex) {
+                            req = null;
+                            return "ERROR: request failed";
+                        });
+        };
+
+        self.create_tmp = function create_tmp() {
+            var dfd = $.Deferred();
```

Comment:
Why is this deferred necessary? Wouldn't it work just to return the promise from cockpit.spawn directly?